regression/apps/scripts/fs.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
+
+# SPDX-License-Identifier: MPL-2.0
+
+set -e
+set -x
+
+check_file_size() {
+    local file_name="$1"
+    local expected_size="$2"
+
+    if [ ! -f "$file_name" ]; then
+        echo "Error: File does not exist."
+        return 1
+    fi
+
+    actual_size=$(du -b "$file_name" | cut -f1)
+
+    if [ "$actual_size" -eq "$expected_size" ]; then
+        return 0
+    else
+        echo "Error: File size is incorrect: expected ${expected_size}, but got ${actual_size}."
+        return 1
+    fi
+}
+
+test_ext2() {
+    local ext2_dir="$1"
+    local test_file="$2"
+
+    cd ${ext2_dir}
+
+    # Test case for the big file feature
+    for i in $(seq 1 10); do
+        truncate -s 500M ${test_file}
+        check_file_size ${test_file} $((500 * 1024 * 1024))
+        truncate -s 2K ${test_file}
+        check_file_size ${test_file} $((2 * 1024))
+    done
+
+    # Clean up
+    rm -f ${test_file}
+    sync
+    cd -
+}
+
+test_fdatasync() {
+    fdatasync/fdatasync /
```

Comment:
Are there any sync-related tests from the gvisor test suite?

Reply:
`fsync` is included in the gvisor test suite, and it works well in Asterinas, but `fdatasync` is not covered. Therefore, I just add a simple `fdatasync` test to ensure the functionality.